src/plot_api/to_image.js

```diff
@@ -0,0 +1,95 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+/*eslint dot-notation: [2, {"allowPattern": "^catch$"}]*/
+
+'use strict';
+
+var Plotly = require('../plotly');
+
+/**
+ * @param {object} gd figure Object
+ * @param {object} opts option object
+ * @param opts.format 'jpeg' | 'png' | 'webp' | 'svg'
+ */
+function toImage(gd, opts) {
+    var Snapshot = require('../snapshot');
```

Comment:
Oh. That mean there's a circular dependency somewhere between `src/plot_api/` and `src/snapshot`. 

Removing that circular dependency might be tricky. I think it would require splitting up `src/plot_api/plot_api.js` into smaller files. Something nobody really wants to do.

You should be able to get away with using `Plotly.Snapshot` in `src/plot_api/to_image.js` without any issues.